library/core/src/ptr/const_ptr.rs

```diff
@@ -1768,6 +1768,47 @@ impl<T> *const [T] {
     }
 }
 
+impl<T, const N: usize> *const [T; N] {
+    /// Returns a raw pointer to the array's buffer.
+    ///
+    /// This is equivalent to casting `self` to `*const T`, but more type-safe.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// #![feature(array_ptr_get)]
+    /// use std::ptr;
+    ///
+    /// let arr: *const [i8; 3] = ptr::null();
+    /// assert_eq!(arr.as_ptr(), ptr::null());
```

Comment:
unimportant nit: The example could be a bit more useful, showing a pointer to an actual array and then maybe passing `as_ptr` to `copy_nonoverlapping` to copy out of the array or something like that.